src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -648,8 +663,11 @@ protected async Task ProcessInvAsync(INetworkPeer peer, List<InventoryVector> in
 
             if (blockHashAnnounced)
             {
-                this.logger.LogDebug("Block was announced, sending getheaders.");
-                await this.ResyncAsync().ConfigureAwait(false);
+                if (this.cachedHeaders.Count == 0)
+                {
+                    this.logger.LogDebug("Block was announced, sending getheaders.");
```

Comment:
I would change the text to be "An Inv was received, requesting new headers"

Also we could put this condition earlier before the `InventoryVector ` foreach

Reply:
Done!